obsei/preprocessor/base_text_cleaner.py

```diff
@@ -0,0 +1,51 @@
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional
+from enum import Enum
+from pydantic import BaseModel, PrivateAttr
+
+
+from obsei.analyzer.base_analyzer import AnalyzerRequest
+
+
+class CleaningFunctions(str, Enum):
+    to_lower_case: str = "to_lower_case"
+    remove_white_space: str = "remove_white_space"
+    remove_punctuation: str = "remove_punctuation"
+    remove_special_characters: str = "remove_special_characters"
+    decode_unicode: str = "decode_unicode"
+    remove_date_time: str = "remove_date_time"
+    replace_domain_keywords: str = "replace_domain_keywords"
+    stem_text: str = "stem_text"
+    remove_stop_words: str = "remove_stop_words"
+
+
+class BaseTextProcessorConfig(BaseModel):
+    TYPE: str = "Base"
+    text_cleaning_functions: Optional[List[CleaningFunctions]]
+    language: Optional[str] = "english"
+
+    class Config:
+        arbitrary_types_allowed = True
+        use_enum_values = True
+
+
+class BaseTextPreprocessor(BaseModel):
+    TYPE: str = "Base"
+    stop_words: Optional[List[str]] = []
+    domain_keywords: Optional[List[Dict]] = []
+    stemmer: Any
```

Comment:
How about moving these to config class -
```
stop_words: Optional[List[str]] = []    
domain_keywords: Optional[List[Dict]] = []
stemmer: Any
```
Also add sane defaults (we can do it in constructor).

Why I am insisting on moving them to config class is because of separating dynamic and static part.
So let's say in large production deployment of Obsei, there will be TextProcessor Node which perform text preprocessing an it will be stateless. Incoming config will bring state to it. So for example -
Same production system is shared between finance domain and legal domain. Both will have different domain_keywords as per current design we need to have two separate `BaseTextPreprocessor` Nodes. Instead if we move it in config then we only need to have one Node and it will perform preprocessing based on input config.

Reply:
I understood this point when the last time you mentioned it, however, my bad that I missed making this change